### DISCUSSION TITLE
How can I disable the microphone permission prompt while playing live streams using the AMS iOS SDK for WebRTC playback?

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5193] [FAQs]
I have successfully integrated the AMS iOS SDK into my application to publish and play live WebRTC streams. However, I am encountering an issue where the SDK prompts for microphone permission when playing the streams. I would like to prevent or disable this prompt specifically during playback.

How can I modify the SDK configuration or implement changes to prevent the microphone permission request while playing streams? I want to ensure that microphone access is not required solely for playback purposes.

**Comment by mekya** [comment]
Yes, it's a known issue in the WebRTC stack and we've fixed this by patching into the WebRTC stack. 
In order to use this patch, we've created an auxiliary method and it's available in the sample as follows. 

```swift
    private func dontAskMicPermissionForPlaying() {
        let webRTCConfiguration = RTCAudioSessionConfiguration.init()
        webRTCConfiguration.mode = AVAudioSession.Mode.moviePlayback.rawValue
        webRTCConfiguration.category = AVAudioSession.Category.playback.rawValue
        webRTCConfiguration.categoryOptions = AVAudioSession.CategoryOptions.duckOthers
                             
        RTCAudioSessionConfiguration.setWebRTC(webRTCConfiguration)
    }
```

If you call this method before any operation in your iOS application, iOS SDK will not ask you for microphone permission while it's playing. 

Cheers

**Reply by Kanaktechtree** [reply]
This is working fine.
Thank you @Mohit-3196 @mekya

**Reply by mekya** [reply]
Thank you for letting us know @Kanaktechtree
